dpc/src/record/payload.rs

```diff
@@ -126,7 +125,9 @@ mod tests {
         // Create a random byte array, construct a payload from it, and check its byte array matches.
         for i in 0..Testnet2::RECORD_PAYLOAD_SIZE_IN_BYTES {
             let expected_payload = (0..i).map(|_| u8::rand(rng)).collect::<Vec<u8>>();
-            let candidate_payload = Payload::<Testnet2>::from(&expected_payload).to_bytes_le().unwrap();
+            let candidate_payload = Payload::<Testnet2>::from(expected_payload.clone())
+                .to_bytes_le()
+                .unwrap();
```

Comment:
What are the guarantees around this call?

Reply:
While I would usually say that I'm just preserving the existing logic, this time I'll note that this is in `mod tests`, so there doesn't need to be any :wink:.

Reply:
Ah, missed the context here 👍